src/twisted/internet/testing.py

```diff
@@ -967,3 +967,54 @@ def createWithCleanup(cls, testInstance: TestCase, publisher: LogPublisher) -> S
         publisher.addObserver(obs)
         testInstance.addCleanup(lambda: publisher.removeObserver(obs))
         return obs
+
+
+def benchmarkWithReactor(test_target):
+    """
+    Decorator for running a benchmark tests that loops the reactor.
+
+    This is designed to decorate test method executed using pytest and
+    pytest-benchmark.
+    """
+
+    @inlineCallbacks
```

Comment:
This should be using `async` functions, there's no need to use inlineCallbacks anymore?

Reply:
yes. It can be migrated ... after many years of writing py2 Twisted code... I just go with inlineCallbacks by default

Reply:
I just pushed this.